hbase-common/src/main/java/org/apache/hadoop/hbase/util/UnsafeAvailChecker.java

```diff
@@ -39,6 +42,10 @@
       @Override
       public Boolean run() {
         try {
+          final Configuration conf = HBaseConfiguration.create();
```

Comment:
We will pass in a Configuration when constructoring a client Connection for accessing HBase cluster, but here we just use the default one, it may lead to strange behavior when user does not use the default config file, or creating the Configuration by programming

Reply:
@Apache9  That's a good point .How about having both with system property (if defined ) taking precedence over the configuration property. That allows clients to  define a config variable in places where they can pass in with the config file in the class path ?